src/helpers/getDatasetState.js

```diff
@@ -0,0 +1,25 @@
+import moment from 'moment'
+// Returns the dataset lifecycle state variables for the UI rendering
+export default (dataset) => {
+  const {
+    expires_on: expiredOn,
+    is_available: isAvailable,
+    is_processed: isProcessed,
+    is_processing: isProcessing,
+    success
+  } = dataset
+
+  const isExpired = moment(expiredOn).isBefore(Date.now())
+  const isComplete = isProcessed && isAvailable && success // successfully processed
+  const isProcessingError = success === false // failed during processing
```

Comment:
I am thinking about renaming this to `hasProcessingError`. I originally matched it with other property names (using the `is` prefix), but I think renaming might be a better fit!

Reply:
i think `isFailed` should work here

Reply:
Great name 👍